pkg/ddl/job_table.go

```diff
@@ -354,7 +354,8 @@ func (d *ddl) startLocalWorkerLoop() {
 func (s *jobScheduler) startDispatchLoop() {
```

Comment:
we should wrap startDispatchLoop inside another loop to avoid transient network error，we can do it later

Reply:
After this PR, if the scheduler runs without the `startDispatchLoop`, does it have side effect to the cluster?

Reply:
All DDLs will be blocked until the cluster elects a new DDL owner.

Reply:
I think we'd better wrap a retry loop for `startDispatchLoop` in this PR. Just afraid the owner is still on this TiDB node and refresh lease succeed.